modules/core/include/opencv2/core/cvdef.h

```diff
@@ -817,9 +817,9 @@ namespace cv
 class hfloat
 {
 public:
-#if CV_FP16_TYPE
+#if CV_FP16_TYPE && defined __ARM_FP16_FORMAT_IEEE
 
-    hfloat() : h(0) {}
+    hfloat() = default;
```

Comment:
Yes, I didn't want to change this code, but `hfloat() : h(0) {}` will cause an error at compile time:
```bash
opencv/modules/core/include/opencv2/core/hal/intrin.hpp:169:1 error: call to implicitly-deleted default constructor of 'union
CV_INTRIN_DEF_TYPE_TRAITS(hfloat, short, ushort, hfloat, float, double, float);
^
opencv/modules/core/include/opencv2/core/hal/intrin.hpp:126:43: note: expanded from macro 'CV_INTRIN_DEF_TYPE_TRAITS'
            union { type l; int_type i; } v; \
                         ^
note: default constructor of '' is implicitly deleted because variant field 'l' has a non-trivial default constructor
```
Any suggestions?